HoloPipelines/pipelines/components/compDicom2numpy.py

```diff
@@ -25,27 +32,21 @@ def loadScan(scanPath):
     return slices
 
 
-def getPixelsHU(scans):
-    image = np.stack([s.pixel_array for s in scans])
-    image = image.astype(np.int16)
-
-    image[image == -2000] = 0
-
-    intercept = scans[0].RescaleIntercept
-    slope = scans[0].RescaleSlope
+def loadPixelArray(inputPath):
+    reader = sitk.ImageSeriesReader()
 
-    if slope != 1:
-        image = slope * image.astype(np.float64)
-        image = image.astype(np.int16)
+    dicomName = reader.GetGDCMSeriesFileNames(inputPath)
+    reader.SetFileNames(dicomName)
 
-    image += np.int16(intercept)
+    image = reader.Execute()
+    numpyArrayImage = sitk.GetArrayFromImage(image)
 
-    return np.array(image, dtype=np.int16)
+    return numpyArrayImage
 
 
 def resample(dataPath, new_spacing=[1, 1, 1]):
     scan = loadScan(dataPath)
-    image = getPixelsHU(scan)
+    image = loadPixelArray(dataPath)
```

Comment:
@UdomkarnBoonyaprasert  This is the only point where I see SITK used. Could you explain how does this replace GDCM? It looks like you were dealing with numpy and the scan object returned from pydicom.

Reply:
Also when you say replace GDCM, does this mean we can skip the external GDCM installation step mentioned in the README.md?

Reply:
If GDCM can be removed, then this file should not import pydicom anymore. You will need to clean up the `loadScan()` function and `scan = loadScan(dataPath)`. Don't leave unused code behind.

Reply:
GDCM is an optional dependency for Pydicom for reading the compressed pixel data. `scan = loadScan(dataPath)` is still needed for reading the individual slice position, and thickness which are used to calculate the resize factor (image re-sampling)

```
        spacing = map(
            float,
            (
                [scan[0].SliceThickness]
                + [scan[0].PixelSpacing[0], scan[0].PixelSpacing[1]]
            ),
        )
        spacing = np.array(list(spacing))
```

Reply:
Thanks for clearing up my misunderstanding. So pydicom can be used without GDCM if you are not dealing with compressed pixel data.

This loc `image = np.stack([s.pixel_array for s in scans])` was the one that required GDCM previously.

ref: https://stackoverflow.com/questions/54657768/pydicom-cant-read-pixel-data-and-needs-gdcm-or-pillow